### ISSUE TITLE
Tag names, descriptions are not available in Mustache templates

### ISSUE THREAD

**Issue by DavidBiesack** [swagger-api/swagger-codegen#5074]
##### Description

The actual name and description of the tags in an OpenAPI document is not preserved in the variables available to templates. Only the *identifier* form is carried over. If a tag contains blanks, they are lost. So it is impossible to use swagger-codegen to create labels for operation tags for -l html such that the tags are rendered as in swagger-ui

Further, swagger-codegen organizes operations by tags ( `{{apis}}` ) so that if an operation has multiple tags, the operation is repeated in the output.

##### Swagger-codegen version

2.2.2-SNAPSHOT

##### Swagger declaration file content or url

[modified-petstore-swagger.json](https://gist.githubusercontent.com/DavidBiesack/70d2910322d5c68d730878f0d8c36587/raw/6b93d6450a74ea08bd31ef0250240e4dcd11a083/modified-petstore-swagger.json?_sm_au_=iVVDq7n0kPL7s5tR) is a simplified sub of petstore, but with the tags renamed to "pet" -> "Personal Pet", "store" -> "Pet Store" and "user" -> "Pet Owner" 

The operation 
```
    "/store/inventory": {
      "get": {
        "tags": [
            "Pet Store", "Personal Pets"
        ],

```

is tagged with two tags (line 271 of the gist), which is meant to indicate that the operation should be groups with both "Pet Store" and "Personal Pets". 

This yields labels for the goups in the html2 navigation (unfortunately ALLCAPS style used  in html2)

API METHODS - PERSONALPET
API METHODS - PETSTORE
API METHODS - PETOWNERS

and the above operation (getInventory) has **duplicate** HTML output: 

file:///tmp/petstore/index.html#api-PersonalPets-getInventory
file:///tmp/petstore/index.html#api-PetStore-getInventory

See [the -debugOperations output](https://gist.githubusercontent.com/DavidBiesack/f635d3504320cba1701f2e35ad9379d3/raw/a391d420ff15f0134fc1f36cb21e13463e066a83/debugOperations.txt?_sm_au_=iVVDq7n0kPL7s5tR) ; note that the original tag text "Personal Pet" or "Pet Store" or "Pet Owner" is not available, nor are the tag descriptions  "Everything about your Pets" or externalDocs

In summary, tags are meant to be used to organize operations and the Open API speci specifically allows multiple tags per operation.

##### Command line used for generation

java -DdebugOperations -jar swagger-codegen-cli.jar  generate \
  -i https://gist.githubusercontent.com/DavidBiesack/70d2910322d5c68d730878f0d8c36587/raw/6b93d6450a74ea08bd31ef0250240e4dcd11a083/modified-petstore-swagger.json?_sm_au_=iVVDq7n0kPL7s5tR \
  -l html2 -o /tmp/petstore

##### Related issues

#4423

##### Suggest a Fix

The proposed fix is to add an `{{allOperations}}' array for use in templates, which is not organized by tags / apis (so one can ensure no duplicates in output). Add a ``{{tag}}` object  in each  `{{api}}`. Within an `{{operation}}` have an array of tag objects so one can use `{{#tags}} {{tag}} {{name}} {{description}} {{/tags}}` etc.

**Comment by wing328** [comment]
I think this is related to https://github.com/swagger-api/swagger-codegen/issues/1073

**Comment by wing328** [comment]
Closed via #5455 (in 2.3.0). Please pull the latest to give it a try.
